File: PROJET_ADOPTE/src/main/java/fr/formation/model/Image.java
package fr.formation.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
@Entity
@Table (name ="image")
public class Image {
	
	@Enumerated(EnumType.ORDINAL)
	protected FormatImage format;
	
	@Column(name = "ima_nom")
	protected String titre;
	
	@Column(name = "ima_contenu")
	protected byte [] contenu;
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ima_id")
	protected int id;
	
	
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public byte[] getContenu() {
		return contenu;
	}
	public void setContenu(byte[] contenu) {
		this.contenu = contenu;
	}
	public String getTitre() {
		return titre;
	}
	public void setTitre(String titre) {
		this.titre = titre;
	}
	public FormatImage getFormat() {
		return format;
	}
	public void setFormat(FormatImage format) {
		this.format = format;
	}
	
	public void Modifier() {
		
	}
	
}

File: angularAdopte/src/app/model/image.ts
import { Observable } from 'rxjs';
export class Image {
  public get contenu(): Uint8Array | undefined {
    return this._contenu;
  }
  public set contenu(value: Uint8Array | undefined) {
    this._contenu = value;
  }
  public get format(): string | undefined {
    return this._format;
  }
  public set format(value: string | undefined) {
    this._format = value;
  }
  public get titre(): string | undefined {
    return this._titre;
  }
  public set titre(value: string | undefined) {
    this._titre = value;
  }
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }
  constructor(
    private _id?: number,
    private _titre?: string,
    private _format?: string,
    private _contenu?: Uint8Array
  ) {}
}
